src/attr/_make.py

```diff
@@ -859,6 +882,52 @@ def _determine_eq_order(cmp, eq, order, default_eq):
     return eq, order
 
 
+def _determine_attrib_eq_order(cmp, eq, order, default_eq):
+    """
+    Validate the combination of *cmp*, *eq*, and *order*. Derive the effective
+    values of eq and order.  If *eq* is None, set it to *default_eq*.
+    """
+    if cmp is not None and any((eq is not None, order is not None)):
+        raise ValueError("Don't mix `cmp` with `eq' and `order`.")
+
+    def decide_callable_or_boolean(value):
+        """
+        Decide whether a key function is used.
+        """
+        if callable(value):
+            value, key = True, value
+        else:
+            key = None
+        return value, key
+
+    # cmp takes precedence due to bw-compatibility.
+    if cmp is not None:
+        warnings.warn(_CMP_DEPRECATION, DeprecationWarning, stacklevel=3)
```

Comment:
I guess we should remove this warning now that the customization is promoting `cmp` back to the center as syntactical sugar?

Reply:
1) Should we "un-deprecate" ``cmp`` from both ``attrib`` and ``attrs``?
2) Do we need the property ``Attribute.cmp``?

Reply:
1. yes
2. I think we do? It's still a derived attribute that can be used as syntactic sugar, to set eq and order at once – agreed?